src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -676,44 +676,39 @@ private static EnumFieldInfo[] TopologicalSortEnumFields(EnumFieldInfo[] enumFie
                 return enumFields;
             }
 
-            var indices = new int[enumFields.Length];
+            var indices = new (int negativePopCount, int index)[enumFields.Length];
             for (int i = 0; i < enumFields.Length; i++)
             {
-                indices[i] = i;
+                // we want values with more bits set to come first so negate the pop count
```

Comment:
```suggestion
                // We want values with more bits set to come first so negate the pop count.
                // Keep the index as a second comparand so that sorting stability is preserved.
```